tests/checkpoint_transaction/run.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+set -eu
+
+cur=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+source $cur/../_utils/test_prepare
+WORK_DIR=$TEST_DIR/$TEST_NAME
+
+function run() {
+	export GO_FAILPOINTS="github.com/pingcap/dm/syncer/checkCheckpointInMiddleOfTransaction=return"
+
+	run_sql_file $cur/data/db1.prepare.sql $MYSQL_HOST1 $MYSQL_PORT1 $MYSQL_PASSWORD1
+	check_contains 'Query OK, 1 row affected'
+
+	# run dm master
+	run_dm_master $WORK_DIR/master $MASTER_PORT $cur/conf/dm-master.toml
+	check_rpc_alive $cur/../bin/check_master_online 127.0.0.1:$MASTER_PORT
+	check_metric $MASTER_PORT 'start_leader_counter' 3 0 2
+
+	# bound source1 to worker1
+	run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER1_PORT
+	dmctl_operate_source create $cur/conf/source1.yaml $SOURCE_ID1
+
+	# start a task in all mode
+	dmctl_start_task_standalone $cur/conf/dm-task.yaml
+
+	# check diff
+	check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+
+	# copy file
+	cp $cur/data/db1.increment1.sql $WORK_DIR/db1.increment1.sql
+	run_sql_file $WORK_DIR/db1.increment1.sql $MYSQL_HOST1 $MYSQL_PORT1 $MYSQL_PASSWORD1
+	sleep 0.1 # wait big_transaction start
```

Comment:
not sure if this will introduce an unstable test. We can inject a sleep to make sure the sleep time will end in a `false` `isTransactionEnd` status.